components/ai_chat/core/browser/ollama/ollama_model_fetcher_unittest.cc

```diff
@@ -0,0 +1,347 @@
+  }
+
+  void ShowModel(const std::string& model_name,
+                 ModelDetailsCallback callback) override {
+    show_model_callbacks_[model_name] = std::move(callback);
+  }
+
+  // Test helpers to trigger callbacks
+  void TriggerFetchModelsCallback(
+      std::optional<std::vector<ModelInfo>> models) {
+    if (fetch_models_callback_) {
+      std::move(fetch_models_callback_).Run(std::move(models));
+    }
+  }
+
+  void TriggerShowModelCallback(const std::string& model_name,
+                                std::optional<ModelDetails> details) {
+    auto it = show_model_callbacks_.find(model_name);
+    if (it != show_model_callbacks_.end()) {
+      std::move(it->second).Run(std::move(details));
+      show_model_callbacks_.erase(it);
+    }
+  }
+
+ private:
+  ModelsCallback fetch_models_callback_;
+  std::map<std::string, ModelDetailsCallback> show_model_callbacks_;
+};
+
+}  // namespace
+
+class OllamaModelFetcherTest : public testing::Test {
+ public:
+  OllamaModelFetcherTest() = default;
+  ~OllamaModelFetcherTest() override = default;
+
+  void SetUp() override {
+    OSCryptMocker::SetUp();
+    prefs::RegisterProfilePrefs(pref_service_.registry());
+    ModelService::RegisterProfilePrefs(pref_service_.registry());
+
+    model_service_ = std::make_unique<ModelService>(&pref_service_);
+
+    shared_url_loader_factory_ =
+        base::MakeRefCounted<network::WeakWrapperSharedURLLoaderFactory>(
+            &test_url_loader_factory_);
+
+    ollama_model_fetcher_ = std::make_unique<OllamaModelFetcher>(
+        *model_service_, &pref_service_, shared_url_loader_factory_);
+
+    // Inject mock OllamaService for testing
+    auto mock = std::make_unique<MockOllamaService>(shared_url_loader_factory_);
+    mock_ollama_service_ = mock.get();
+    ollama_model_fetcher_->ollama_service_ = std::move(mock);
+  }
+
+  void TearDown() override { OSCryptMocker::TearDown(); }
+
+  ModelService* model_service() { return model_service_.get(); }
+  OllamaModelFetcher* ollama_model_fetcher() {
+    return ollama_model_fetcher_.get();
+  }
+  network::TestURLLoaderFactory* test_url_loader_factory() {
+    return &test_url_loader_factory_;
+  }
+  sync_preferences::TestingPrefServiceSyncable* pref_service() {
+    return &pref_service_;
+  }
+  MockOllamaService* mock_ollama_service() { return mock_ollama_service_; }
+
+ private:
+  base::test::TaskEnvironment task_environment_;
+  sync_preferences::TestingPrefServiceSyncable pref_service_;
+  std::unique_ptr<ModelService> model_service_;
+  network::TestURLLoaderFactory test_url_loader_factory_;
+  scoped_refptr<network::SharedURLLoaderFactory> shared_url_loader_factory_;
+  std::unique_ptr<OllamaModelFetcher> ollama_model_fetcher_;
+  raw_ptr<MockOllamaService> mock_ollama_service_;
+};
+
+TEST_F(OllamaModelFetcherTest, FetchModelsAddsNewModels) {
+  size_t initial_count = model_service()->GetModels().size();
+
+  // Trigger FetchModels
+  ollama_model_fetcher()->FetchModels();
```

Comment:
with above callback, this will be put after all the `EXPECT_CALL`

Reply:
Fixed in 5db664f6e2640df6f1f59439dcfaa1d4b7ffd7a8